File: HLC01_03Java/src/main/java/org/example/juegoRol03/Fireball.java
package org.example.juegoRol03;

import java.util.Random;

public class Fireball implements Spell {
    private final String name = "fuego";

    @Override
    public String getName() {
        return name;
    }

    @Override
    public void cast(Personaje caster, Personaje target) {
        Random random = new Random();
        int damage = random.nextInt(10, 21); // Random between 10 (inclusive) and 21 (exclusive)
        target.receiveDamage(damage);
    }

}

File: HLC01_03Kotlin/src/main/kotlin/juegoRol03/Fireball.kt
package org.example.juegoRol03

import kotlin.random.Random

class Fireball : Spell {
    override val name: String = "fuego"

    override fun cast(caster: Personaje, target: Personaje) {
        val damage = Random.nextInt(10, 21)
        target.receiveDamage(damage)
    }
}